supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/EndpointParser.java

```diff
@@ -0,0 +1,219 @@
+package de.tum.cit.endpointanalysis;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.expr.ArrayInitializerExpr;
+import com.github.javaparser.ast.expr.Expression;
+import com.github.javaparser.ast.expr.NormalAnnotationExpr;
+import com.github.javaparser.ast.expr.SingleMemberAnnotationExpr;
+
+public class EndpointParser {
+
+    static final String ENDPOINT_PARSING_RESULT_PATH = "endpoints.json";
+
+    static final String REST_CALL_PARSING_RESULT_PATH = "restCalls.json";
+
+    private static final Logger logger = LoggerFactory.getLogger(EndpointParser.class);
+
+    public static void main(String[] args) {
+        final String relativeDirectoryPath = ".." + File.separator + ".." + File.separator + "src" + File.separator + "main" + File.separator + "java";
+        final Path absoluteDirectoryPath = Paths.get(relativeDirectoryPath).toAbsolutePath().normalize();
```

Comment:
`Path.of` can handle the separators for you.
```suggestion
        final Path absoluteDirectoryPath = Path.of("../../src/main/java").toAbsolutePath().normalize();
```